File: sdks/java/connect-sdk/src/main/java/com/mobiscroll/connect/models/WebhookSubscribeResponse.java
package com.mobiscroll.connect.models;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

/** Result of {@code Webhooks.subscribeWebhook}. */
public final class WebhookSubscribeResponse {

    private final boolean success;
    private final String provider;
    private final WebhookSubscription subscription;
    private final String serverWebhookUrl;
    private final String channelId;

    @JsonCreator
    public WebhookSubscribeResponse(
            @JsonProperty("success") boolean success,
            @JsonProperty("provider") String provider,
            @JsonProperty("subscription") WebhookSubscription subscription,
            @JsonProperty("serverWebhookUrl") String serverWebhookUrl,
            @JsonProperty("channelId") String channelId) {
        this.success = success;
        this.provider = provider;
        this.subscription = subscription;
        this.serverWebhookUrl = serverWebhookUrl;
        this.channelId = channelId;
    }

    public boolean isSuccess() { return success; }
    public String getProvider() { return provider; }
    public WebhookSubscription getSubscription() { return subscription; }
    /** The Connect-hosted URL registered with the provider to receive change notifications. */
    public String getServerWebhookUrl() { return serverWebhookUrl; }
    /** Same value as {@link WebhookSubscription#getChannelId()}, surfaced at the top level for convenience. */
    public String getChannelId() { return channelId; }
}

File: sdks/dotnet/src/Mobiscroll.Connect/Models/WebhookSubscribeResponse.cs
namespace Mobiscroll.Connect.Models;

/// <summary>The <c>subscription</c> object returned by <see cref="Resources.Webhooks.SubscribeWebhookAsync"/>.</summary>
public sealed class WebhookSubscription
{
    public string ChannelId { get; set; } = string.Empty;

    /// <summary>Present for Google.</summary>
    public string? ResourceId { get; set; }

    /// <summary>ISO 8601 timestamp; present for some providers.</summary>
    public string? Expiration { get; set; }
}

public sealed class WebhookSubscribeResponse
{
    public bool Success { get; set; }
    public string Provider { get; set; } = string.Empty;
    public WebhookSubscription Subscription { get; set; } = new();
    public string ServerWebhookUrl { get; set; } = string.Empty;
    public string ChannelId { get; set; } = string.Empty;
}
